3-UsingAPIs/Week2/homework/ex3-rollAnAce.js

```diff
@@ -10,27 +11,25 @@
 // ! Do not change or remove the next two lines
 const rollDice = require('../../helpers/pokerDiceRoller');
 
-function rollDiceUntil(wantedValue) {
+async function rollDiceUntil(wantedValue) {
   // TODO: rewrite this function using async/await
-  const recurse = () => {
-    return rollDice().then((value) => {
-      if (value !== wantedValue) {
-        return recurse();
-      }
-      return value;
-    });
-  };
-  return recurse();
+  const value = await rollDice()
+  while (value !== wantedValue) {
+    return rollDiceUntil("ACE")
+  }
+  return value
 }
 
 // TODO refactor this function to use try/catch
-function main() {
-  rollDiceUntil('ACE')
-    .then((results) => console.log('Resolved!', results))
-    .catch((error) => console.log('Rejected!', error.message));
+async function main() {
+  try {
+    const response = await rollDiceUntil('ACE')
+    const results = await response
```

Comment:
`response` is already the wanted result. So there is not need for this statement. And in particular, there is no need for `await` because `response` is not a promise.